tests/test_es/test_es.py

```diff
@@ -60,44 +60,35 @@ def test_es_types(init_interpreter, doc_type):
 
 
-# @pytest.mark.parametrize("doc_type", [dt.parser.doc_type for dt in init_interpreter().translators.values()])
-# def test_get_list_from_path(init_interpreter, doc_type):
-#     if doc_type == 'file':
-#         return
-#     interpreter = init_interpreter
-#     items = items_in_file(doc_type)[2]
-#     parser = interpreter.translators[doc_type].parser
-#     names = get_names(parser)
-#
-#     fails = []
-#     for item in items:
-#         submitter_id = item['submitter_id']
-#         results = get_item_from_elasticsearch(parser.name, doc_type, submitter_id)
-#         result_length = len(results)
-#         if result_length != 1:
-#             if result_length < 1:
-#                 fails.append('Not exist expected {doc_type} with submitter_id {item} in ES'
-#                              .format(doc_type=doc_type, item=submitter_id))
-#             else:
-#                 fails.append('Duplicated {doc_type} with submitter_id {item} in ES'
-#                              .format(doc_type=doc_type, item=submitter_id))
-#             continue
-#         result = results[0]
-#         for name in names:
-#             value = result.__getattr__(name) if name in result else None
-#
-#             path = get_path_by_name(parser, name)
-#
-#             fn = path["fn"]
-#
-#             table_list = get_table_list_from_path(parser, doc_type, path["path"])
-#             table_list_in_order = order_table_list(table_list)
-#
-#             sql = generate_sql_from_table_list(table_list_in_order, fn, name, submitter_id)
-#             val = execute_sql_query(sql)
-#
-#             if value != val:
-#                 fails.append('{doc_type} with submitter_id={item} has field {name} '
-#                              'with {val} in database and {value} in ES'
-#                              .format(doc_type=doc_type, item=submitter_id, name=name, val=val, value=value))
-#     assert fails == []
+@pytest.mark.parametrize("doc_type", [dt.parser.doc_type for dt in init_interpreter().translators.values()])
+def test_get_list_from_path(init_interpreter, doc_type):
+    if doc_type == 'file':
+        return
+    interpreter = init_interpreter
+    items = items_in_file(doc_type)[2]
+    parser = interpreter.translators[doc_type].parser
+    names = get_names(parser)
+
+    fails = []
+    for item in items:
+        submitter_id = item['submitter_id']
+
+        results = ESValue(parser, submitter_id, doc_type, names)
+
+        result_length = results.length
+
+        if result_length != 1:
+            if result_length < 1:
+                fails.append('Not exist expected {doc_type} with submitter_id {item} in ES'
+                             .format(doc_type=doc_type, item=submitter_id))
+            else:
+                fails.append('Duplicated {doc_type} with submitter_id {item} in ES'
+                             .format(doc_type=doc_type, item=submitter_id))
+
+        value = AggregatorValue(parser, submitter_id, doc_type, names)
```

Comment:
We still must regenerate the same SQL Query many times many of them only have the difference is passing value. Can we improve this by caching SQL statements as a formatted string and just pass the value into the formatted string.

Reply:
Could you address the comment above?